### PR TITLE
openthread: add support for OpenThread linking from library

### PR DESCRIPTION
**Pull request by MarekPorwisz**
OpenThread supports certification by inheritance. This allows
application to be Thread certified by linking certified library instead
of building OpenThread from sources.
This change integrates linking Thread from the library instead of
building it from zephyrs port sources. Included libraries are not yet
certified.

Signed-off-by: Marek Porwisz <marek.porwisz@nordicsemi.no>

### CODE REVIEW

File: openthread/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+include(${NRFXLIB_DIR}/common.cmake)
+nrfxlib_calculate_lib_path(lib_path)
+set(OT_LIB_PATH ${CMAKE_CURRENT_SOURCE_DIR}/${lib_path})
+
+if (NOT CONFIG_OPENTHREAD_BUILD_SOURCES)
+  if(NOT EXISTS ${OT_LIB_PATH})
+    message(WARNING "This platform is not certified by OpenThread yet.
```
Comment (lmaciejonczyk) [inline openthread/CMakeLists.txt]: Shouldn't it be rather something like: "Missing OT library path, fix it or compile OT from source"?

File: openthread/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  elseif(CONFIG_OPENTHREAD_MTD)
+    zephyr_library_import(
+      openthread-mtd
+      ${OT_LIB_PATH}/libopenthread-mtd.a
+    )
+  endif()
+
+  zephyr_include_directories(include)
+
+endif()
```
Comment (lmaciejonczyk) [inline openthread/CMakeLists.txt]: missing newline at the end of file

File: openthread/Kconfig [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+# Empty for now.
```
Comment (lmaciejonczyk) [inline openthread/Kconfig]: Do we really need empty files?

**Review by lmaciejonczyk — COMMENTED** [review]
LGTM

File: openthread/CMakeLists.txt (hunk 1 above)
Comment (MarekPorwisz) [reply openthread/CMakeLists.txt]: ${OT_LIB_PATH} is set automatically by NCS build scripts to path pointing to the correct platform (e.g lib/cortex-m4/soft-float for nRF52840). All libraries shall be provided by Nordic.
The idea here is that user does not update those libraries. If the library is not available this means this platform is not certified or is not supported(making it not certified as well)

**Pull request closed without merging**